sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,34 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   */
+  object SubstituteHint extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
```

Comment:
The current grammar allows multiple query hints in a query. What will happen if we do that? For example:
```SQL
SELECT /*+ MAPJOIN(tbl_a, tbl_a) */
       *
FROM   tbl_a A
       JOIN tbl_b B
        ON B.id = A.id
       JOIN (SELECT /*+ MAPJOIN(tbl_c) */ 
                         XA.id
                  FROM   tbl_b XA
                          LEFT SEMI JOIN tbl_c XB
                          ON XB.id = XA.id) C
        ON C.id = A.id
```